status/status.go

```diff
@@ -49,7 +51,22 @@ func NewHealthCheckHandler(checkSource status.HealthCheckSource, sharedSecret re
 }
 
 func (h *healthHandlerImpl) ServeHTTP(w http.ResponseWriter, req *http.Request) {
-	metadata, newHealthStatusCode := h.computeNewHealthStatus(req)
+	if sharedSecret := h.healthCheckSharedSecret.CurrentString(); sharedSecret != "" {
+		token, err := httpserver.ParseBearerTokenHeader(req)
+		if err != nil {
+			errors.WriteErrorResponse(w, errors.NewUnauthorized(wparams.NewSafeParam("message", err.Error())))
```

Comment:
generally conjure errors should be strongly typed and not include arbitrary messages like this. If we want to log more detail, it should happen server-side. I'd just do an empty `Unauthorized` here and `PermissionDenied` below once we have a token to check.

Reply:
Addressed, though I think this will make debugging harder.